File: backend/app/schemas/earnings.py
"""收益看板相关 Pydantic schemas。"""

from datetime import datetime
from typing import Optional

from pydantic import BaseModel


class EarningsSummary(BaseModel):
    """收益汇总。"""
    pending_balance: str  # 记账余额（待提现）
    withdrawn_total: str  # 已提现总额
    available_balance: str  # 可用余额（= 记账余额 - 已冻结提现金额）


class EarningsRecord(BaseModel):
    """收益明细条目。"""
    id: int
    amount: str
    type: str  # first_reward / followup_reward / team_bonus / recommend / sale_commission
    source_user_id: Optional[int] = None
    source_email: Optional[str] = None
    business_id: str
    created_at: datetime


class EarningsListResponse(BaseModel):
    """收益明细列表响应。"""
    summary: EarningsSummary
    records: list[EarningsRecord]
    total: int

File: user-web/src/services/earnings.ts
import { request } from "./api";

// ── 后端 EarningsSummary: { pending_balance, withdrawn_total, available_balance }（均为 str）──
export interface EarningsSummary {
  pending_balance: string;
  withdrawn_total: string;
  available_balance: string;
}

// ── 后端 EarningsRecord ──
export interface EarningsRecord {
  id: number;
  amount: string;
  type: string;
  source_user_id: number | null;
  source_email: string | null;
  business_id: string;
  created_at: string;
}

// ── 后端 EarningsListResponse: { summary, records, total } ──
export interface EarningsListResponse {
  summary: EarningsSummary;
  records: EarningsRecord[];
  total: number;
}

export interface EarningsListParams {
  type?: string;
  limit?: number;
  offset?: number;
}

export const earningsApi = {
  /** 收益汇总 + 明细，单端点返回二者（后端无独立 /summary /records）。 */
  getEarnings: (params?: EarningsListParams) =>
    request<EarningsListResponse>({
      method: "GET",
      url: "/users/me/earnings",
      params,
    }),
};
